controllers/jobs.js

```diff
@@ -1,19 +1,70 @@
+const Job = require("../models/Job");
+const { BadRequestError, NotFoundError } = require("../errors");
+const { StatusCodes } = require("http-status-codes");
+
 const getAllJobs = async (req, res) => {
-  res.send("get all jobs");
+  const jobs = await Job.find({ createdBy: req.user.userId }).sort("createdAt");
+  res.status(StatusCodes.OK).json({ jobs, count: jobs.length });
 };
 
 const getJob = async (req, res) => {
-  res.send("get job");
+  console.log(req.user, req.params);
+  const {
+    user: { userId },
+    params: { id: jobId },
+  } = req;
+  console.log("printing data", jobId, userId);
+  const job = await Job.findOne({ _id: jobId, createdBy: userId });
```

Comment:
Nitpick (deson't require changes):
I'd recommend checking if `req.params.id` or **any other form of stringified id** is a valid ObjectID by using mongoose's [isValidObjectId utility function](https://mongoosejs.com/docs/5.x/docs/api.html#mongoose_Mongoose-isValidObjectId) BEFORE using it within a `Job` method. Otherwise, if you supply an invalid id, then... for example, `Job.findOne` may throw a casting error:
```
err: CastError: Cast to ObjectId failed for value "1" (type string) at path "_id" for model "Job"
```
This should also apply to your other controllers that use an `id` to find, delete, or update a `Job`.